code/statsd/src/main/scala/com/avast/datadog4s/statsd/metric/TimerImpl.scala

```diff
@@ -44,7 +45,14 @@ class TimerImpl[F[_]: Sync](
   }
 
   override def record(duration: Duration, tags: Tag*): F[Unit] =
-    F.delay {
-      statsDClient.recordExecutionTime(aspect, duration.toMillis, sampleRate, (tags ++ defaultTags): _*)
+    timerMode match {
```

Comment:
Wouldn't it be more efficient to have two different implementations of Timer? That way we only need to do pattern matching once on init time, and not every time the metric is updated.

Reply:
yep, i already have those changes ready ;)